### ISSUE TITLE
Rounding the corners of bars

### ISSUE THREAD

**Issue by ricburton** [danielgindi/Charts#1066]
<img width="616" alt="screen shot 2016-05-23 at 17 58 03" src="https://cloud.githubusercontent.com/assets/438160/15488915/f225cf8c-210f-11e6-9ecf-1a270b6a2170.png">

What would be the best way to achieve this effect? I’m guessing I’ll need to modify the way Charts draws in some way.

**Comment by liuxuan30** [comment]
I have been investigated on this a few months ago. It's possible, however, we need to change how we create the bar rect.

**Comment by liuxuan30** [comment]
That's what I investigated before, you can try first. I am still looking for other options.

in `public func drawDataSet:` of bar chart renderer
```swift
                let cornerRadius: CGFloat = CGRectGetWidth(barRect) <= 5 ? 1.0 : 2.0
                let bezierPath = UIBezierPath(roundedRect: barRect, byRoundingCorners: UIRectCorner.AllCorners, cornerRadii: CGSizeMake(cornerRadius, cornerRadius))
                
                let roundedPath = bezierPath.CGPath
                
                // if drawing the bar shadow is enabled
                if (drawBarShadowEnabled)
                {
                    barShadow.origin.x = barRect.origin.x
                    barShadow.origin.y = viewPortHandler.contentTop
                    barShadow.size.width = barRect.size.width
                    barShadow.size.height = viewPortHandler.contentHeight
                    
                    CGContextSetFillColorWithColor(context, dataSet.barShadowColor.CGColor)
//                    CGContextFillRect(context, barShadow)
                    CGContextAddPath(context, roundedPath)
                    CGContextFillPath(context)
                }
                
                // Set the color for the currently drawn value. If the index is out of bounds, reuse colors.
                CGContextSetFillColorWithColor(context, dataSet.colorAt(j).CGColor)
//                CGContextFillRect(context, barRect)
                CGContextAddPath(context, roundedPath)
                CGContextFillPath(context)
```

**Comment by ricburton** [comment]
Thanks so much for the feedback! I’ll play with this.

**Comment by liuxuan30** [comment]
yeah I don't think it's a great one, especially your bars are very thin

**Comment by FahadIqbal** [comment]
its not working can anyone help me to do it ?

**Comment by CoderZyc** [comment]
@liuxuan30请问下，这个在最新的版本还可以？

**Comment by FahadIqbal** [comment]
Thanks for your dearness

**Comment by liuxuan30** [comment]
@CoderZyc  not merged yet. there is a PR for it.

**Comment by acegreen** [comment]
Hey @liuxuan30 this looks awesome, any ETA on this?

**Comment by liuxuan30** [comment]
not really. I don't think it's perfect, and not having time to investigate other options.

**Comment by acegreen** [comment]
@liuxuan30 no worries, there is another solution that worked out nicely #1917

**Comment by liuxuan30** [comment]
@acegreen basically they work the same way. Just wrap up nicely. Since more people asking, I am reopening this topic

**Comment by kirti0525** [comment]
Hi,
Can you please help me with adding vertical dashed lines in between bars in bar graph and making the top corner of the bar slightly curvy . I am uploading the screenshot for the sample.
![screenshot_2018-09-06 zeplin - project](https://user-images.githubusercontent.com/40993277/45160690-c25a1980-b207-11e8-90b2-87e673494521.png)

**Comment by lalkrishna** [comment]
Any update on this?

**Comment by jacobsapps** [comment]
It looks like this and a similar pull request - https://github.com/danielgindi/Charts/pull/1917 - have been dead for at least 2 years. Is it worth throwing away all the work they took?

**Comment by kelvinofula** [comment]
Is there a way to achieve the same with the slices of a piechart?

**Comment by scinfu** [comment]
+1

**Comment by JanC** [comment]
+1

**Comment by 4np** [comment]
If #4297 was merged, you could just override the renderer.

**Comment by scinfu** [comment]
Not merged :/

**Comment by AppCodeZip** [comment]
Recently I have been working on a project, to which I have added a chart library and display a beautiful rounded bar chart in swift. We can achieve this functionality easily. Please check this [demo](https://www.appcodezip.com/2021/03/rounded-bars-in-iOS-Charts.html).
![Screenshot 2021-03-27 at 7 42 42 PM](https://user-images.githubusercontent.com/65787178/112728905-81ee5280-8f4f-11eb-9ab6-4be70232b512.png)

**Comment by nunocaseiro** [comment]
+1

**Comment by Eramirez06** [comment]
when this will be merge?

**Comment by iOSLeadGautam** [comment]
Hi All,

I want to get a Barchart with rounded bar corners  and pie having rounded corners for every section.
Waiting for this PR to be merged.

Any advise would be appreciated. 

Also can someone suggest me how to get pie chart like added image. 


![PieChart_Requirement](https://user-images.githubusercontent.com/122786516/212654586-5f29e7cb-b37d-4b3c-8f93-b64f61b3ab41.png)

**Comment by BenjaminAlbrectsen** [comment]
Is it still not possible to round the corners? https://github.com/danielgindi/Charts/issues/1066#issuecomment-223186364 does not work for me, since I'm using SPM

**Comment by pmairoldi** [comment]
Currently already implemented [here](https://github.com/ChartsOrg/Charts/blob/1bad5469f57628782110b05996d4fa00473abf06/Source/Charts/Data/Implementations/Standard/BarChartDataSet.swift#L54-L78)
Closing in favour of #5250 for full platform support
